### ISSUE TITLE
Loading into arrays/maps drops head comment (regression)

### ISSUE THREAD

**Issue by neiser** [mikefarah/yq#1445]
**Describe the bug**
Loading into arrays/maps from files having a head comment drops the head comment. This is a regression as it has worked in yq 4.27.2 but I noticed it stopped working in yq 4.30.4. I was not able to identify the exact commit yet which introduced this bug. 

Version of yq: 4.27.2 (works), 4.30.4 (does not work)
Operating system: linux
Installed via: binary release (old version), snap (new version)

**Input Yaml**
cats.yml:
```yaml
# my cats
- cat1
- cat2
```

animals.yml:
```yaml
- cats.yaml
```

**Command**
The command you ran:
```
yq '.[] |= load(.)' animals.yaml
```

**Actual behavior**

yq 4.30.4 (latest release) outputs unexpected behavior (head comment `# my cats` is gone):

```yaml
- - cat1
  - cat2
```

**Expected behavior**

yq 4.27.2 outputs expected behavior:

```yaml
- # my cats
  - cat1
  - cat2
```
**Additional context: Also fails with maps**

Note than when the file is loaded into a map:
```yaml
animals: cats.yaml
```
using `yq '.animals |= load(.)' animals.yaml`, the bug also appears with yq 4.30.4:

output:
```yaml
animals:
  - cat1
  - cat2
```

**Additional context: Foot comments are preserved**

The bug does not appear when `cats.yaml` contains a foot comment:
```
- cat1
- cat2
# end of cats
```

Output of `yq '.animals |= load(.)' animals.yaml`:
```yaml
animals:
  - cat1
  - cat2
  # end of cats
```

**Post scriptum**

This regression is a blocker for us as you can imagine that in reality, we do have a lot of animals (and many more cats than just 2), so preserving comments is really required to keep the herd under control :smile_cat:

@mikefarah Also, let me take the chance to thank you for that excellent tool you made!

**Comment by mikefarah** [comment]
More than 2 cats?!? Love it. Will fix in next release (away for the weekend, so early next week)

**Comment by mikefarah** [comment]
Fixed in 4.30.5
